### PR TITLE
Create Gherkin tests for device-status-reachability and device-status-roaming

### PR DESCRIPTION
**Pull request by mdomale**
Feature file for device-reachability-status & device-roaming-status

#### What type of PR is this?

* tests

#### What this PR does / why we need it:

Gherkin syntax feature file for syntax

#### Which issue(s) this PR fixes:

Fixes #

#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

**Comment by bigludo7** [comment]
Hello @mdomale 
Thanks for the contribution.
Probably better to split in 2 files to have one file for device-reachability-status and another for device-connectivity-status.

On the file itself we should align with the [API-Testing-Guidelines](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Testing-Guidelines.md).  We should have environment variables, background section, etc... You can probably take a look on this [PR](https://github.com/camaraproject/DeviceLocation/pull/189) that could be used as an illustration of the guidelines.

**Comment by mdomale** [comment]
Updated as per newly agreed camara testing guidelines

File: code/Test_definitions/device-reachability-status.feature [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    When request reachability status for "$.phonenumber" is set to a invalid testing device not available by the service
+    Then response code is 404
+	And the response property "$.status" is 404
+    And the response property "$.code" is "NOT_FOUND"
+    And the response property "$.message" contains a user friendly text
+	
+ @Device_Reachability_Status_07_ConnectedSMS_Get_Method_call
+   Scenario: Check device reachability synchronously if the device is connected for SMS usage & invalid method 
+    Given Use BaseURL
+    When get method request reachability status for "$.phonenumber"
```
Comment (bigludo7) [inline code/Test_definitions/device-reachability-status.feature:90]: and Operation is GET

**Review by bigludo7 — COMMENTED** [review]
Thanks for the contribution.

Few suggestions for your appreciation. I did the review for code/Test_definitions/device-reachability-status.feature but I guess same applies for the second API.

- Add Test cases for 403 `INVALID_TOKEN_CONTEXT` 

```  @...._device_token_mismatch
  Scenario: Inconsistent access token context for the device
    # To test this, a token have to be obtained for a different device
    Given the request body property "$.device" is set to a valid testing device
    And the header "Authorization" is set to a valid access token emitted for a different device
    When the HTTP "POST" request is sent
    Then the response status code is 403
    And the response property "$.status" is 403
    And the response property "$.code" is "INVALID_TOKEN_CONTEXT"
    And the response property "$.message" contains a user friendly text
```

- Add Test Case for 422 for DEVICE_IDENTIFIERS_MISMATCH
```   @..._device_identifiers_mismatch
  Scenario: Device identifiers mismatch
    # To test this, at least 2 types of identifiers have to be provided, e.g. a phoneNumber and the IP address of a device associated to a different phoneNumber
    Given that config_var "identifier_types_unsupported" contains at least 2 items
    And the request body property "$.device" includes several identifiers, each of them identifying a valid but different device
    When the HTTP "POST" request is sent
    Then the response status code is 422
    And the response property "$.status" is 422
    And the response property "$.code" is "DEVICE_IDENTIFIERS_MISMATCH"
    And the response property "$.message" contains a user friendly text
```

- Add Test Case for 422 for DEVICE_NOT_APPLICABLE
```   @...._device_not_supported
  Scenario: Service not available for the device
    Given that service is not supported for all devices commercialized by the operator 
    And the request body property "$.device" is set to a valid device for which the service is not applicable
    When the HTTP "POST" request is sent
    Then the response status code is 422
    And the response property "$.status" is 422
    And the response property "$.code" is "DEVICE_NOT_APPLICABLE"
    And the response property "$.message" contains a user friendly text
```

- Add Test Case for 422 for UNABLE_TO_PROVIDE_REACHABILITY_STATUS

- We can add one for `406` but not very valuable - I'm fine if you skip this one.

**Comment by bigludo7** [comment]
@mdomale @akoshunyadi 
It looks very good for me.

For me oly one thing missing: all  explicit error are not tested. You have added 422 `DEVICE_IDENTIFIERS_MISMATCH`. We should also add test cased for 422 `DEVICE_NOT_APPLICABLE`, 422 `UNABLE_TO_PROVIDE_REACHABILITY_STATUS` and 422 `UNSUPPORTED_DEVICE_IDENTIFIERS`

Then for me we will be good !
Thanks

**Comment by mdomale** [comment]
@bigludo7 @sachinvodafone PR amended. Kindly review

**Review by bigludo7 — DISMISSED** [review]
Thanks @mdomale 
I saw you added the 3 422 so it is good for me !

File: code/Test_definitions/device-reachability-status.feature [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    And the request body property "$.device" is set to a valid testing device supported by the service
+    When the request "getReachabilityStatus" is sent
+    Then the response code is 200
+    And the response header "Content-Type" is "application/json"
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response body complies with the OAS schema at "/components/schemas/ReachabilityStatusResponse"
+    And the response property "$.status" is 200
+    And the response property "$.code" is "OK"
+    And the response property "$.reachabilityStatus" is  "NOT_CONNECTED"
+
```
Comment (akoshunyadi) [inline code/Test_definitions/device-reachability-status.feature:55]: We are sending the same request and expect once it returns SMS, DATA or not connected. I think it should be in the precondition, whether the device is connected to SMS DATA or not connected.

**Comment by mdomale** [comment]
@bigludo7  @bigludo7  PR updated .

**Review by akoshunyadi — APPROVED** [review]

**Review by bigludo7 — APPROVED** [review]
lgtm

**Pull request merged**